### PR TITLE
Add registry addon feature for docker on mac/windows

### PR DESCRIPTION
**Pull request by alonyb**
This PR is for #7535

Only in Mac/Windows on KIC this is the output.

```
Rubens-MacBook-Pro:ruben-minikube-act rubendario$ out/minikube start --addons=registry  --vm-driver=docker
😄  minikube v1.9.2 on Darwin 10.15.3
✨  Using the docker driver based on user configuration
👍  Starting control plane node minikube in cluster minikube
🚜  Pulling base image ...
🔥  Creating Kubernetes in docker container with (CPUs=2) (6 available), Memory=1998MB (1998MB available) ...
🐳  Preparing Kubernetes v1.18.0 on Docker 19.03.2 ...
    ▪ kubeadm.pod-network-cidr=10.244.0.0/16
E0416 20:24:39.187173   21722 kubeadm.go:319] Overriding stale ClientConfig host https://127.0.0.1:32812 with https://127.0.0.1:32820
❗  This container is having trouble accessing https://k8s.gcr.io
💡  To pull new external images, you may need to configure a proxy: https://minikube.sigs.k8s.io/docs/reference/networking/proxy/
🌟  Enabling addons: default-storageclass, registry, storage-provisioner
💡  Registry addon on with docker uses 32821 please use that instead of default 5000
📘  For more information see: https://minikube.sigs.k8s.io/docs/drivers/docker
🏄  Done! kubectl is now configured to use "minikube"
```

Output:
Screen Shot 2020-04-16 at 20 31 16

### CODE REVIEW

File: pkg/addons/addons.go [hunk 1]
```diff
@@ -176,6 +176,16 @@ https://github.com/kubernetes/minikube/issues/7332`, out.V{"driver_name": cc.Dri
 		return nil
 	}
 
+	if name == "registry" {
+		port, err := machine.RegistryPort(cc)
+		if err != nil {
+			errors.Wrap(err, "Can't get Registry port")
```
Comment (medyagh) [inline pkg/addons/addons.go:182]: change this to, errors.Wrap(err, "registery port")

File: pkg/addons/addons.go [hunk 2]
```diff
@@ -176,6 +176,16 @@ https://github.com/kubernetes/minikube/issues/7332`, out.V{"driver_name": cc.Dri
...
+		}
+		if runtime.GOOS != "linux" {
```
Comment (medyagh) [inline pkg/addons/addons.go:184]: change this to 
```
if drivers.IsKIC(cc.Driver) && runtime.GOOS != "linux"
```

File: pkg/addons/addons.go [hunk 3]
```diff
@@ -176,6 +176,16 @@ https://github.com/kubernetes/minikube/issues/7332`, out.V{"driver_name": cc.Dri
...
+			out.T(out.Notice, `This is the port you should use for the registry addon "{{.port}}"`, out.V{"port": port})
```
Comment (medyagh) [inline pkg/addons/addons.go:185]: change this wording to:
`   registry addon on with {{.driver}} uses {{.port}} please use that
for more information see
"https://minikube.sigs.k8s.io/docs//link to documentation"

(and add a documentaiton to the website)

File: pkg/drivers/kic/kic.go [hunk 4]
```diff
@@ -203,6 +208,15 @@ func (d *Driver) GetSSHPort() (int, error) {
 	return p, nil
 }
 
+// GetRegistryAddonPort returns port for use with registry
+func GetRegistryAddonPort(cc *config.ClusterConfig) (int, error) {
+	p, err := oci.ForwardedPort(cc.Driver, cc.Name, constants.RegistryAddonPort)
```
Comment (medyagh) [inline pkg/drivers/kic/kic.go:213]: this should use ForwardedPort
only if the goos. is NOT linux and also IF driver is Kic
use 
driver.IsKIC(cc.Driver) to check if driver is KIC

Comment (medyagh) [reply pkg/drivers/kic/kic.go:213]: no need to have this function here, just handle it in the same place u call it. bellow

File: site/content/en/docs/drivers/docker.md [hunk 5]
```diff
@@ -19,19 +19,13 @@ The Docker driver allows you to install Kubernetes into an existing Docker insta
 
 ## Known Issues
 
-- On macOS, containers might get hung and require a restart of Docker for Desktop. See [docker/for-mac#1835](https://github.com/docker/for-mac/issues/1835)
```
Comment (medyagh) [inline site/content/en/docs/drivers/docker.md:22]: why did u delete this one, I think you wanted to update the line bellow.

Comment (alonyb) [reply site/content/en/docs/drivers/docker.md]: I restore that file

**Comment by medyagh** [comment]
make sure to run `make lint` and fix all the lint issues

File: pkg/addons/addons.go [hunk 6]
```diff
@@ -176,6 +177,17 @@ https://github.com/kubernetes/minikube/issues/7332`, out.V{"driver_name": cc.Dri
 		return nil
 	}
 
+	if name == "registry" {
+		if driver.IsKIC(cc.Driver) && runtime.GOOS != "linux" {
+			port, err := kic.GetRegistryAddonPort(cc)
+			if err != nil {
+				errors.Wrap(err, "registry port")
+			}
+			out.T(out.Notice, `Registry addon on with {{.driver}} uses {{.port}} please use that.
```
Comment (medyagh) [inline pkg/addons/addons.go:186]: u can use 		out.T(out.Documentation, 

for better icon

**Comment by medyagh** [comment]
can you also share the output of minikube addon eanble registry

that one should also add notice

**Comment by medyagh** [comment]
@alonyb can you please run make lint to fix all the lint issues ?

u can also see them in github acitons https://github.com/kubernetes/minikube/pull/7603/checks?check_run_id=579470326

**Comment by alonyb** [comment]
> @alonyb can you please run 'make lint' and then fix all the lint issues ?
> 
> u can also see them in github acitons https://github.com/kubernetes/minikube/pull/7603/checks?check_run_id=579470326

sure, now looks perfect!

**Comment by medyagh** [comment]
@alonyb please make sure the integration test is running the test on mac os for this.

**Pull request merged**